core/vm/evm.go

```diff
@@ -515,29 +515,30 @@ func getOrComputeTobinTaxFunctionSelector() []byte {
 
 // TobinTransfer performs a transfer that takes a tax from the sent amount and gives it to the reserve
 func (evm *EVM) TobinTransfer(db StateDB, sender, recipient common.Address, gas uint64, amount *big.Int) (leftOverGas uint64, err error) {
-	reserveAddress, err := GetRegisteredAddressWithEvm(params.ReserveRegistryId, evm)
 
-	if err != nil && err != errors.ErrSmartContractNotDeployed && err != errors.ErrRegistryContractNotDeployed {
-		log.Trace("Error in tobin transfer", "error", err)
-		return gas, err
-	}
-
-	if amount.Cmp(big.NewInt(0)) != 0 && err == nil {
-		ret, gas, err := evm.Call(AccountRef(sender), *reserveAddress, getOrComputeTobinTaxFunctionSelector(), gas, big.NewInt(0))
-		if err != nil {
-			return gas, err
+	if amount.Cmp(big.NewInt(0)) != 0 {
+		reserveAddress, err := GetRegisteredAddressWithEvm(params.ReserveRegistryId, evm)
+		if err != nil && err != errors.ErrSmartContractNotDeployed && err != errors.ErrRegistryContractNotDeployed {
+			log.Trace("Error in tobin transfer", "error", err)
 		}
 
-		// Expected size of ret is 64 bytes because getOrComputeTobinTax() returns two uint256 values,
-		// each of which is equivalent to 32 bytes
-		if binary.Size(ret) == 64 {
-			numerator := new(big.Int).SetBytes(ret[0:32])
-			denominator := new(big.Int).SetBytes(ret[32:64])
-			tobinTax := new(big.Int).Div(new(big.Int).Mul(numerator, amount), denominator)
+		if err == nil {
+			ret, gas, err := evm.Call(AccountRef(sender), *reserveAddress, getOrComputeTobinTaxFunctionSelector(), gas, big.NewInt(0))
```

Comment:
This should be moved into a `contract_comm` function that returns the tobin tax

Reply:
unfortunately an import cycle makes this very difficult.